syntax = "proto3";
package PointCloudTools;

import "google/protobuf/empty.proto";

import "common.proto";
import "metrics.proto";
import "filters.proto";
import "segmentation.proto";
import "reconstruct.proto";

service FilterServices {
    rpc applySubsetFilter(subsetFilterRequest) returns (stringResponse) {}
}

service SegmentationServices {
    rpc applySegmentation(segmentationRequest) returns (stringResponse) {}
}

service MetricServices {
    rpc computeMetric (metricServiceRequest) returns (numericResponse) {}
}

/* PHOTO RECON */

message ReconstructImageRequest {
    string images_path = 1; // Image locations
    ReconstructOptions options = 2;
    
}
message ReconstructImageResponse {
    string message = 1;
}
service PhotogrammetryServices {
    rpc reconstructFromImages(ReconstructImageRequest) returns (ReconstructImageResponse) {}
    rpc getJobStatus(google.protobuf.Empty) returns (JobStatusResponse) {}
}